src/hdskins/java/com/voxelmodpack/hdskins/skins/NetClient.java

```diff
@@ -34,20 +79,52 @@ public NetClient putFile(String key, String contentType, URI file) {
         return this;
     }
 
+    /**
+     * Sets the headers to be included with this request.
+     * @param headers   Headers to send
+     *
+     * @return itself for chaining
+     */
     public NetClient putHeaders(Map<String, ?> headers) {
         this.headers = headers;
 
         return this;
     }
 
-    public MoreHttpResponses send() throws IOException {
+    /**
+     * Commits and sends the request.
+     */
+    private void send() throws IOException {
```

Comment:
You should keep the response separated from the actual client. If you want to keep a reference to the return value, then just save it in a variable.

Reply:
NetClient is intended to wrap the request/response process as a whole. Granted, I suppose it doesn't matter now that you've gutted it.

Reply:
If you're familiar with requests libraries in python and javascript, this is more or less the way they operate. Call a function with the method, address, headers, and payload, get a response object back. The object usually has other functions to get the content as text, json, bytes, etc.

Reply:
I also don't really want to get into the situation where I have to dispose of the client, and then dispose the request. It seems so much cleaner just to have one object.

Plus, previously NetClient was supposed to lazy load the request. So you could just do `client.getResponseCode()` without going through any intermediaries.

Reply:
You added a method that resets the client, so I assume you weren't going to get rid of it anytime soon.

Reply:
Reset method was added with the eventual objective of being able to re-use the client based on what you were doing with clients in Valhalla.